specs/OAuth/AuthManager.md

```diff
@@ -0,0 +1,891 @@
+
+| Name | Description |
+|-|-|
+| AuthRequestParams(String, String) | Construct AuthRequestParams object with `responseType` and `clientId` parameters passed in the order mentioned. |
+| AuthRequestParams(String, String, Windows.Foundation.Uri) | Construct AuthRequestParams object with `responseType`, `clientId` and a `redirectUri` passed in the order mentioned. |
+
+
+## AuthRequestParams Methods
+
+| Name | Description | Parameters | Returns |
+|-|-|-|-|
+| CreateForAuthorizationCodeRequest(String) | Helper method to create for an authorization code grant request ("code" response type) with required parameters. | String `clientId` | AuthRequestParams |
+| CreateForAuthorizationCodeRequest(String, Windows.Foundation.Uri) | Helper method to create for an authorization code grant request ("code" response type) with required parameters. | String `clientId` , Windows.Foundation.Uri `redirectUri` | AuthRequestParams |
+| CreateForImplicitRequest(String) | Helper method to create for an implicit grant request ("token" response type) with required parameters. | String `clientId` | AuthRequestParams |
+| CreateForImplicitRequest(String, Windows.Foundation.Uri) | Helper method to create for an implicit grant request ("token" response type) with required parameters. | String `clientId` , Windows.Foundation.Uri `redirectUri` | AuthRequestParams |
+
+## AuthRequestParams Properties
+
+| Name | Description | Type |
+|-|-|-|
+| ResponseType | Specifies the required "response_type" parameter of the authorization request described by the section 3.1.1 of [RFC 6749](https://www.rfc-editor.org/rfc/rfc6749.html#section-3.1.1). | String |
+| ClientId | Specifies the required "client_id" parameter of the authorization request described by the section 2.2 of [RFC 6749](https://www.rfc-editor.org/rfc/rfc6749.html#section-2.2). | String |
+| RedirectUri | Specifies the optional "redirect_uri" parameter of the authorization request described by the section 3.1.2 of [RFC 6749](https://www.rfc-editor.org/rfc/rfc6749.html#section-3.1.2). | Windows.Foundation.Uri |
+| State | Specifies the recommended "state" parameter of the authorization request described in section 4.1.1 of [RFC 6749] (https://www.rfc-editor.org/rfc/rfc6749#section-4.1.1) | String |
+| Scope | Specifies the optional "scope" parameter of the authorization request described by the section 3.3 of [RFC 6749](https://www.rfc-editor.org/rfc/rfc6749#section-3.3). | String |
+| CodeChallenge | Used as the PKCE code_challenge described by the section 4.2 of [RFC 7636](https://www.rfc-editor.org/rfc/rfc7636#section-4.2). | String |
+| CodeChallengeMethod | Specifies the optional "code_challenge_method" parameter of the authorization request described by the section 4.3 of [RFC 7636](https://www.rfc-editor.org/rfc/rfc7636#section-4.3). | CodeChallengeMethodKind |
+| AdditionalParams | Additional parameters passed along in the query string of the request URL. | Windows.Foundation.Collections.IMap<String, String> |
```

Comment:
Don't break this out. This is part of the request URL you already get from most oauth services when registering your app. I would ditch this class all-together.

Reply:
![image](https://github.com/user-attachments/assets/957cebf6-84cf-4d26-a435-a10ec6d54be0)
In this case, we can keep this class.

Reply:
Keeping this as we are creating a comprehensive list. We are adding two new APIs for simplication.
InitiateRequestAsync(Uri fullAuthorizeURI, Uri RedirectURI)
InitiateRequestAsyncWithoutCallback(Uri fullAuthorizeUri)

We will provide examples for these two APIs at the top of spec doc.